src/PerfView/PerfViewData.cs

```diff
@@ -4120,9 +4121,54 @@ protected internal override StackSource OpenStackSourceImpl(string streamName, T
                 pinnedObjectAnalyzer.Execute(GCPinnedObjectViewType.PinnedObjectAllocations);
             }
             else if (streamName == "CCW Ref Count")
-            {
-                // TODO use the callback model.  We seem to have an issue getting the names however. 
-                foreach (var data in events.ByEventType<CCWRefCountChangeTraceData>())
+            {                
+                eventSource.Interop.AddCallbackForEvents<TaskCCWQueryRuntimeClassNameArgs>(args =>
+                {
+                    sample.Metric = 1;
+                    sample.TimeRelativeMSec = args.TimeStampRelativeMSec;
+                    var stackIndex = stackSource.GetCallStack(args.CallStackIndex(), args);
+
+                    var ccwRefKindName = "CCW QueryRuntimeClassName";
+                    var ccwRefKindNameIndex = stackSource.Interner.FrameIntern(ccwRefKindName);
+                    stackSource.Interner.CallStackIntern(ccwRefKindNameIndex, stackIndex);
+
+                    sample.StackIndex = stackIndex;
+                    stackSource.AddSample(sample);
+                });
+                eventSource.Interop.AddCallbackForEvents<TaskCCWRefCountIncArgs>(args=> {
+                    sample.Metric = 1;
+                    sample.TimeRelativeMSec = args.TimeStampRelativeMSec;
+                    var stackIndex = stackSource.GetCallStack(args.CallStackIndex(), args);
+
+                    var ccwRefKindName = "CCW RefCountInc";
+                    var ccwRefKindNameIndex = stackSource.Interner.FrameIntern(ccwRefKindName);
+                    stackSource.Interner.CallStackIntern(ccwRefKindNameIndex, stackIndex);
+
+                    var ccwRefCount = "CCW NewRefCnt " + args.refCount;
+                    var ccwRefCountIndex = stackSource.Interner.FrameIntern(ccwRefCount.ToString());
+                    stackSource.Interner.CallStackIntern(ccwRefCountIndex, stackIndex);
+
+                    sample.StackIndex = stackIndex;
+                    stackSource.AddSample(sample);
+                });
+                eventSource.Interop.AddCallbackForEvents<TaskCCWRefCountDecArgs>(args =>
+                {
+                    sample.Metric = 1;
```

Comment:
I believe we want the metric to be -1 so that decs 'cancel' the incs.   

It really would be useful to have the class name associated with the CCW as well.  We really should update the event to include this, but that is a separate issue.